HTMLSerializer.js

```diff
@@ -347,6 +413,21 @@ var HTMLSerializer = class {
   }
 
   /**
+   * Create a function that will generate unique strings which can be used as
+   * ids.
+   *
+   * @return {Function} A funtion that generates a unique string each time it is
+   *     called.
+   */
+  generateIDGenerator() {
```

Comment:
What if the document already contains an id with this value? Do you need to track existing ids in the document?